owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsPictureUploadsFragment.kt

```diff
@@ -121,7 +121,9 @@ class SettingsPictureUploadsFragment : PreferenceFragmentCompat() {
                 prefPictureUploadsSourcePath?.summary = DisplayUtils.getPathWithoutLastSlash(it.sourcePath.toUri().path)
                 prefPictureUploadsOnWifi?.isChecked = it.wifiOnly
                 prefPictureUploadsOnCharging?.apply {
-                    isVisible = it.wifiOnly
+                    isEnabled = it.wifiOnly
+                    summary =
+                        if (!it.wifiOnly) getString(R.string.prefs_camera_upload_on_charging_summary) else getString(R.string.prefs_camera_upload_on_charging_summary_disable)
```

Comment:
I wouldn't show a summary when the preference is enabled, as the user knows that it's been enabled only when activating the Wi-Fi preference. Similarly to "Biometric lock", that way we keep consistency in settings 😃